src/python/pants/goal/stats_aggregator.py

```diff
@@ -80,6 +98,8 @@ def __call__(
         if not finished:
             return
 
+        output_contents = ""
```

Comment:
I had the same thought, and was actively looking for if this could have the potential to explode, but I think the number of lines are rather deterministic and never unbounded so in this case I think it would be mostly about style..?

Reply:
Switched to using a list!